go/example.go

```diff
@@ -96,47 +90,55 @@ func main() {
 	client, err := flight.NewClientWithMiddleware(
 		net.JoinHostPort(config.Host, config.Port),
 		nil,
-		[]flight.ClientMiddleware{
-			flight.NewClientCookieMiddleware(),
-		},
+		[]flight.ClientMiddleware{flight.NewClientCookieMiddleware()},
 		grpc.WithTransportCredentials(creds),
 	)
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer client.Close()
 
+	flightClient := &implementations.GoFlightClient{Client: client}
```

Comment:
Let's summarize somewhere why do we actually need to make the wrap?
Mocking original Client interface from arrow-go/flight is not possible?

Reply:
I think we probably could yeah. But if I use the original I'll have to have implementations for all methods in the orginal Client interface and there are some methods that we don't need. What do you think?